src/spawn_blocking.rs

```diff
@@ -0,0 +1,38 @@
+use crate::{
+    concurrency_limit::ConcurrencyLimit, error::Error, make_future_cancellable,
+    ComputeHeavyFutureExecutor,
+};
+
+pub(crate) struct SpawnBlockingExecutor {
+    concurrency_limit: ConcurrencyLimit,
+}
```

Comment:
I wonder if this should actually hold onto a handle—you could initialize it from `Runtime::current` at creation time.

Reply:
The benefit of that being, it would ensure that the spawn_blocking happens in the originating context instead of calling, in case of multiple runtimes? Or something else?